src/uct/api/uct.h

```diff
@@ -1659,4 +1749,263 @@ UCT_INLINE_API ucs_status_t uct_ep_fence(uct_ep_h ep, unsigned flags)
+ * @param [in]    tag_mask  Mask which specifies what bits of the tag to
+ *                          compare.
+ * @param [in]    iov       Points to an array of @ref ::uct_iov_t structures.
+ *                          The @a iov pointer must be valid address of an array
+ *                          of @ref ::uct_iov_t structures. A particular structure
+ *                          pointer must be valid address. NULL terminated pointer
+ *                          is not required.
+ * @param [in]    iovcnt    Size of the @a iov data @ref ::uct_iov_t structures
+ *                          array. If @a iovcnt is zero, the data is considered empty.
+ *                          @a iovcnt is limited by @ref uct_iface_attr_cap_tag_recv_iov
+ *                          "uct_iface_attr::cap::tag::max_iov"
+ * @param [inout] ctx       Context associated with this particular tag, “priv” field
+ *                          in this structure is used to track the state internally.
+ *
+ * @return UCS_OK                - The tag is posted to the transport.
+ * @return UCS_ERR_NO_RESOURCE   - Could not start the operation due to lack of
+ *                                 resources.
+ * @return UCS_ERR_EXCEEDS_LIMIT - No more room for tags in the transport.
+ */
+UCT_INLINE_API ucs_status_t uct_iface_tag_recv_zcopy(uct_iface_h iface,
+                                                     uct_tag_t tag,
+                                                     uct_tag_t tag_mask,
+                                                     const uct_iov_t *iov,
+                                                     size_t iovcnt,
+                                                     uct_tag_context_t *ctx)
+{
+    return iface->ops.iface_tag_recv_zcopy(iface, tag, tag_mask, iov, iovcnt, ctx);
+}
+
+
+/**
+ * @ingroup UCT_TAG
+ * @brief Cancel a tag previously posted by @ref uct_iface_tag_recv_zcopy.
+ *
+ * The tag would be either matched or canceled, in a bounded time, regardless
+ * of the peer actions. The original completion callback of the tag would be
+ * called with the status if @a force is not set.
+ *
+ * @param [in]  iface     Interface to cancel the tag on.
+ * @param [in]  ctx       Tag context which was used for posting the tag. If
+ *                        force is 0, @a ctx->completed_cb will be called with
+ *                        either UCS_OK which means the tag was matched and data
+ *                        received despite the cancel request, or
+ *                        UCS_ERR_CANCELED which means the tag was successfully
+ *                        canceled before it was matched.
+ * @param [in]  force     Whether to report completions to ctx->completed_cb. If
+ *                        nonzero, the cancel is assumed to be successful, and
+ *                        the callback is not called.
+ *
+ * @return UCS_OK  -      The tag is canceled in the transport.
+ */
+UCT_INLINE_API ucs_status_t uct_iface_tag_recv_cancel(uct_iface_h iface,
```

Comment:
no need, send operations could be canceled on the protocol level (unlike receives)